### ISSUE TITLE
[BFA] Item level is wrong

### ISSUE THREAD

**Issue by Cofibib** [arkayenro/arkinventory#1033]
When choosing to display or sort by item level the computed item level is wrong for some items.
Sometimes higher than the one in the tooltip, sometimes lower.

**Comment by Cofibib** [comment]
![wowscrnshot_051918_111224](https://user-images.githubusercontent.com/29673664/40267015-cdf4e01c-5b55-11e8-86d7-2c4dc287c6c2.jpg)
![wowscrnshot_051918_111229](https://user-images.githubusercontent.com/29673664/40267016-ce0ca4d6-5b55-11e8-8cd5-3303aec1d3a8.jpg)

Version BFA r718

**Comment by arkayenro** [comment]
i havent found an item with the wrong item level yet so its a bit hard to fix at the moment

**Comment by Cofibib** [comment]
If you can tell me where to look at, I may try to add some debug information in the code to help find the source.

**Comment by arkayenro** [comment]
got into edit mode
click on an item
select debug info
screenshot that

**Comment by arkayenro** [comment]
could you change your client language to english and see if that fixes the issue

im just wondering in the single quote in the item level tooltip text is causing an issue

**Comment by Cofibib** [comment]
Version BFA r719

![wowscrnshot_052518_195821](https://user-images.githubusercontent.com/29673664/40559220-4c9224de-6056-11e8-9286-8012052c3191.jpg)

Switching to English fix the issue.
Going back to french make the issue back. I tried to clear the WoW Cache folder, clear the bag data without success.
The single quote is also present in live and it's working fine. Also some items are good.
You take the item level from the tooltip ? No API call give this information ? 
But will be strange that the quote in the tooltip convert 217 to 280, I will more expect to have an LUA error or a completely different value.

**Comment by Cofibib** [comment]
The tooltip parsing seems to not work at all:

![wowscrnshot_052518_205214](https://user-images.githubusercontent.com/29673664/40561537-e4742ca0-605d-11e8-80d3-6b6fcdc5cbab.jpg)

To get hits I've uncomented this :

`
if info.osd.upgradeid > 0 or info.osd.bonusids then
			
	-- upgradable or has bonusId that may not adjust the itemlevel return value (eg 615/timewarped), so get item level from tooltip
			
	--ArkInventory.TooltipSetHyperlink( ArkInventory.Global.Tooltip.Scan, info.h )
	ArkInventory.TooltipSetHyperlink( ArkInventory.Global.Tooltip.Scan, info.h )
	local _, _, ilvl = ArkInventory.TooltipFind( ArkInventory.Global.Tooltip.Scan, ArkInventory.Localise["WOW_TOOLTIP_ITEM_LEVEL"], false, true, true, 4, true )
	info.ilvl = tonumber( ilvl ) or info.ilvl
			
	--if info.ilvl ~= info.info[4] then 
		ArkInventory.Output( h, " / ", info.info[4], " / ", ilvl, " / ", info.osd.upgradeid, " / ", info.osd.bonusids )
	--end
			
end
`

The 'ArkInventory.Localise["WOW_TOOLTIP_ITEM_LEVEL"]' seems to give the right answer [^Niveau d'objet (%d+)$]

**Comment by Cofibib** [comment]
I think I got it.
The issue comes from the quote like you suppose.

In the function "ArkInventory.TooltipFind"
The tooltip is "cleaned" using 
```LUA
txt = ArkInventory.TooltipCleanText( obj:GetText( ) )
```
who seems to remove the quote.
But the text to find is not, so it keep the quote.

By adding
```LUA
	TextToFind = ArkInventory.TooltipCleanText(TextToFind)
```
Before starting to loop in the tooltip line it's now working (ArkInventoryTooltip.lua line 411).

But I don't know how it works on live...

**Comment by arkayenro** [comment]
item level can be extracted via the API but upgrade and bonus IDs can change it, so when an item has either of those i have to scan the tooltip instead as thats the only way to easily work out what the bonus IDs have done to it.

in the case of that scepter, its base ilvl is 280, so either the upgrade id or the bonus id is lowering it to 217 (which i've never seen before, typically they increase it, but this is beta)

basically the single quote would have made the tooltip search fail so it reverted to the API itemlevel without any bonus

thanks for find it

**Comment by arkayenro** [comment]
fixed in r720-alpha
